languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/model/JavaModelInterfaceRenderer.kt

```diff
@@ -168,7 +148,17 @@ class JavaModelInterfaceRenderer @Inject constructor(override val vrapTypeProvid
             |public void setValue(String key, ${this.type.toVrapType().simpleName()} value);
             """.trimMargin()
         } else {
-            "public void set${this.name.upperCamelCase()}(final ${this.type.toVrapType().simpleName()} ${this.name.lowerCamelCase()});"
+            return if(this.type is ArrayType){
+                val arrayType = this.type as ArrayType
```

Comment:
Do we really have to say `as arrayType` here? I would have expected that kotlin could infer the type here?

Reply:
"smart cast to ArrayType is impossible, because this.type is a mutable property that could have been changed by this time" - this is the error I get without explicit casting.

Reply:
then you should extract `this.type` outside the if block.